lib/jekyll/filters.rb

```diff
@@ -345,19 +347,12 @@ def sort_input(input, property, order)
 
     private
     def time(input)
-      case input
-      when Time
-        input.clone
-      when Date
-        input.to_time
-      when String
-        Time.parse(input) rescue Time.at(input.to_i)
-      when Numeric
-        Time.at(input)
-      else
+      date = Liquid::Utils.to_date(input)
+      unless date.respond_to?(:strftime)
```

Comment:
Don't want to be nitpicking but could we instead check for `:to_time` ? Because ultimately that's what is going to be invoked on the `date` object.